types/leaflet/index.d.ts

```diff
@@ -1297,13 +1321,13 @@ export class GeoJSON<P = any> extends FeatureGroup<P> {
     /**
      * Adds a GeoJSON object to the layer.
      */
-    addData(data: geojson.GeoJsonObject): Layer;
+    addData(data: geojson.GeoJsonObject): this;
```

Comment:
I'm not sure if a `this` return type is useful here (or at all). What is the reasoning behind this change?

Reply:
(This applies to the below suggestion as well)

I mostly made the change because the [Leaflet documentation](https://leafletjs.com/reference-1.7.1.html#geojson-adddata) specifies that `addData` and `resetStyle` return `this`.

But more importantly, without the changes, it's unclear that these functions return `this`. When the return type is just `Layer`, it's unclear what exactly is being returned. Because these typings sometimes specify `this` as a return type (eg. [`LatLngBounds.extend`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/884526c90d6f3d9dbedbad8fc6841e2b87e3e4ca/types/leaflet/index.d.ts#L162)), `Layer` implies that these functions don't return `this`. I'd expect them to return some other `Layer` object instead.

Knowing that the return type is `this` is useful for callers who want to chain functions and for those extending `GeoJSON` who need to return `this` in overridden functions.